### PR TITLE
Fail vstest task if minimum specified tests not executed

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
Fail vstest task if minimum specified tests are not executed

### CODE REVIEW

File: Tasks/VsTestV2/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -69,6 +69,10 @@
   "loc.input.help.configuration": "Build configuration against which the tests should be reported. If you have defined a variable for configuration in your build task, use that here.",
   "loc.input.label.publishRunAttachments": "Upload test attachments",
   "loc.input.help.publishRunAttachments": "Opt in/out of publishing run level attachments.",
+  "loc.input.label.failOnMinTestsNotRun": "Fail task if a minimum number of tests are not run",
+  "loc.input.help.failOnMinTestsNotRun": "Selecting this option will fail the task if specified minimum number of tests is not run.",
+  "loc.input.label.minimumExpectedTests": "Minimum # of tests",
+  "loc.input.help.minimumExpectedTests": "Specify the minimum # of tests that should be executed to not fail the task.",
```
[the remark this answers is not in the document]
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/Strings/resources.resjson/en-US/resources.resjson]: Also the line:

Total tests executed is calculated as the sum of passed, failed and aborted tests.

File: Tasks/VsTestV2/inputdatacontract.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export interface TestReportingSettings {
     TestResultsDirectory : string;
     TestRunSystem : string;
     TestSourceSettings : TestSourceSettings;
+    ExecutionStatusSettings: MinimumTestsExecutionStatusSettings;
```
Comment (ShreyasRmsft) [inline Tasks/VsTestV2/inputdatacontract.ts]: this goes under execution settings. same change in the c# side

File: Tasks/VsTestV2/inputparser.ts [hunk 3]
```diff
@@ -181,6 +181,23 @@ function getTestReportingSettings(inputDataContract : idc.InputDataContract) : i
 
         inputDataContract.TestReportingSettings.TestRunTitle = `TestRun_${definitionName}_${buildOrReleaseName}`;
     }
+
+    const actionOnThresholdNotMet = tl.getInput('failOnMinTestsNotRun');
```
Comment (ShreyasRmsft) [inline Tasks/VsTestV2/inputparser.ts]: use getBoolInput

File: Tasks/VsTestV2/inputparser.ts [hunk 4]
```diff
@@ -181,6 +181,23 @@ function getTestReportingSettings(inputDataContract : idc.InputDataContract) : i
...
+    {
+        inputDataContract.TestReportingSettings.ExecutionStatusSettings = <idc.MinimumTestsExecutionStatusSettings>{};
+        inputDataContract.TestReportingSettings.ExecutionStatusSettings.ActionOnThresholdNotMet = "fail";
+        
+        const miniumExpectedTests = parseInt(tl.getInput('minimumExpectedTests'));
+        if (!isNaN(miniumExpectedTests)) {
+            inputDataContract.TestReportingSettings.ExecutionStatusSettings.MinimumExecutedTestsExpected = miniumExpectedTests;
+            console.log(tl.loc('minimumExpectedTests', inputDataContract.TestReportingSettings.ExecutionStatusSettings.MinimumExecutedTestsExpected));
+        } else {
+            tl.warning(tl.loc('invalidMinimumExpectedTests'));
```
Comment (ShreyasRmsft) [inline Tasks/VsTestV2/inputparser.ts]: error and fail the task here

Comment (ShreyasRmsft) [inline Tasks/VsTestV2/inputparser.ts]: print the invalid input

File: Tasks/VsTestV2/inputdatacontract.ts (hunk 2 above)
Comment (vagisha-nidhi) [reply Tasks/VsTestV2/inputdatacontract.ts]: This has been added in TestReportingSettings only in c# as well.

File: Tasks/VsTestV2/inputparser.ts (hunk 4 above)
Comment (vagisha-nidhi) [reply Tasks/VsTestV2/inputparser.ts]: Done

Comment (vagisha-nidhi) [reply Tasks/VsTestV2/inputparser.ts]: Done

File: Tasks/VsTestV2/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
[the remark this answers is not in the document]
Comment (vagisha-nidhi) [reply Tasks/VsTestV2/Strings/resources.resjson/en-US/resources.resjson]: Addressed

File: Tasks/VsTestV2/task.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
     "version": {
         "Major": 2,
         "Minor": 160,
```
Comment (ShreyasRmsft) [inline Tasks/VsTestV2/task.json]: change to 161.0

**Pull request merged**